### PR TITLE
GH-41590: [Java] Improve BaseRepeatedValueVector function on isEmpty and isNull operations

### LINKED ISSUE #41590: [Java] Improve `BaseRepeatedValueVector` function on `isEmpty` and `isNull` operations
### Describe the enhancement requested

Referring to `ListVector` base class `BaseRepeatedValueVector` at https://github.com/apache/arrow/blob/main/java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java#L358-L367, the `isEmpty` and `isNull` returns a default value which is rather misleading. Since the sub-class contains the functionality, it would be better to make it abstract or throw as discussed [here](https://github.com/apache/arrow/pull/41285#discussion_r1593798045).

### Component(s)

Java

### PR DESCRIPTION
**Pull request by tlm365**
### Rationale for this change
Resolves #41590 .

### What changes are included in this PR?
Make `abstract` on `isNull` and `isEmpty` of `BaseRepeatedValueVector`.

### Are these changes tested?
Existing tests pass.

### Are there any user-facing changes?
No.

* GitHub Issue: #41590

### CODE REVIEW

**Review by vibhatha — APPROVED** [review]
LGTM! Thanks for working on this.

**Comment by vibhatha** [comment]
@tlm365 could you please rename the PR title to 
`GH-41590: [Java] Improve BaseRepeatedValueVector function on isEmpty and isNull operations`

**Comment by vibhatha** [comment]
cc @lidavidm

File: java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java [hunk 1]
```diff
@@ -355,16 +355,12 @@ public int getInnerValueCountAt(int index) {
             offsetBuffer.getInt(index * OFFSET_WIDTH);
   }
 
-  /** Return if value at index is null (this implementation is always false). */
+  /** Return if value at index is null. */
   @Override
-  public boolean isNull(int index) {
-    return false;
-  }
+  public abstract boolean isNull(int index);
```
Comment (lidavidm) [inline java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java:360]: isNull is just part of the ValueVector interface, so we should be able to get rid of this declaration entirely.

Comment (vibhatha) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java:360]: Yes that works better I guess.

**Review by lidavidm — APPROVED** [review]

**Pull request merged**